source/docs/software/commandbased/commands-v3/mechanisms.rst

```diff
@@ -0,0 +1,434 @@
+# Mechanisms
+
+Mechanisms (called "Subsystems" in Commands v2) represent independently-controlled groupings of robot hardware. They serve two key purposes:
+
+1. **Encapsulation**: Hide hardware implementation details from the rest of your code
+2. **Resource Management**: Ensure only one command uses a mechanism at a time
+
+.. note::
+   If you're familiar with Commands v2 Subsystems, Mechanisms work almost identically. The main differences are the name change and a few API updates for v3's coroutine-based commands.
+
+## Creating a Mechanism
+
+Mechanisms extend the ``Mechanism`` base class from the ``org.wpilib.commands3`` package.
+
+.. code-block:: java
+
+   import org.wpilib.commands3.Mechanism;
+   import org.wpilib.commands3.Command;
+
+   public class Drivetrain extends Mechanism {
+     private final WPI_TalonFX leftLeader = new WPI_TalonFX(1);
+     private final WPI_TalonFX rightLeader = new WPI_TalonFX(2);
+     private final DifferentialDriveOdometry odometry;
+     private final Encoder leftEncoder;
+     private final Encoder rightEncoder;
+
+     public Drivetrain() {
+       // Configure hardware
+       leftLeader.setInverted(false);
+       rightLeader.setInverted(true);
+
+       leftEncoder = new Encoder(0, 1);
+       rightEncoder = new Encoder(2, 3);
+
+       odometry = new DifferentialDriveOdometry(
+         getHeading(),
+         getLeftDistance(),
+         getRightDistance()
+       );
+
+       // Set a default command
+       setDefaultCommand(
+         runRepeatedly(() -> tank(0, 0))
+           .withPriority(Command.LOWEST_PRIORITY)
+           .named("Drive[IDLE]")
+       );
+     }
+
+     // Private hardware control methods
+     private void tank(double left, double right) {
+       leftLeader.set(left);
+       rightLeader.set(right);
+     }
+
+     private void stop() {
+       tank(0, 0);
+     }
+
+     public void resetEncoders() {
+       leftEncoder.reset();
+       rightEncoder.reset();
+     }
+
+     public double getLeftDistance() {
+       return leftEncoder.getDistance();
+     }
+
+     public double getRightDistance() {
+       return rightEncoder.getDistance();
+     }
+
+     public Rotation2d getHeading() {
+       // Implementation depends on your gyro
+       return new Rotation2d();
+     }
+
+     @Override
```

Comment:
down below it sets up periodic stuff differently, and I don't see a periodic method in the javadocs for Mechanism.